### PR TITLE
Moved filters to appropriate packages.

### PR DESCRIPTION
**Pull request by Grebosz**
Fixed small issues in filters (with alpha).
Refactored BlurFIlter to be extend of ConvolutionFIlter. (they shared the same code)

### CODE REVIEW

File: jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java [hunk 1]
```diff
@@ -39,14 +39,8 @@ protected void setPixel(int x, int y, Pixel pixel) {
...
-            if(sourceColors[i] > 255){
-                sourceColors[i] = 255;
-            }
-            if(sourceColors[i] < 0){
-                sourceColors[i] = 0;
-            }
+            sourceColors[i] = Math.min(Math.max(sourceColors[i], 0), 255);
         }
 
         output.setPixel(x, y, sourceColors);
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java]: @" //todo alpha support, investigate when and where alpha chanel is present" havent we already done that?

File: jis-api/src/main/java/pl/edu/uj/JImageStream/model/StreamableImage.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
     //todo extract to IOImageUtilsClass, maybe
     public StreamableImage(File file) throws IOException {
-        bufferedImage = Util.convert(ImageIO.read(file), BufferedImage.TYPE_INT_ARGB);
+        bufferedImage = ImageUtils.convert(ImageIO.read(file), BufferedImage.TYPE_INT_ARGB);
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/model/StreamableImage.java]: delete all todos and delete all useless methods in this class

File: jis-documentation/src/test/java/BlueFilterTest.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import org.junit.Test;
```
Comment (jgprogramming) [inline jis-documentation/src/test/java/BlueFilterTest.java]: could we move all filter test to appropriate package like pl.edu.uj.JImageStream.tests.filters ?

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/BlueFilter.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package pl.edu.uj.JImageStream.filters;
+package pl.edu.uj.JImageStream.filters.color;
 
 import pl.edu.uj.JImageStream.api.core.Filter;
 import pl.edu.uj.JImageStream.model.Pixel;
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/BlueFilter.java]: weren't we supposed to change behavior of blue/red/green filters to filter out chosen color instead of leaving only one chosen?

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/GrayScaleFilter.java [hunk 5]
```diff
@@ -8,6 +8,7 @@
     public void apply(int x, int y) {
         Pixel input = getPixel(x, y);
         int gray = (input.getRed() + input.getBlue() + input.getGreen()) / 3;
-        setPixel(x, y, new Pixel(gray, gray, gray, 255));
+        int alpha = getPixel(x, y).getAlpha();
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/GrayScaleFilter.java]: setPixel(x, y, new Pixel(gray, gray, gray, input.getAlpha())); ?

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/SepiaFilter.java [hunk 6]
```diff
@@ -10,7 +10,8 @@ public void apply(int x, int y) {
         int outputRed = (int) ((input.getRed() * .393) + (input.getGreen() * .769) + (input.getBlue() * .189));
         int outputGreen = (int) ((input.getRed() * .349) + (input.getGreen() * .686) + (input.getBlue() * .168));
         int outputBlue = (int) ((input.getRed() * .272) + (input.getGreen() * .534) + (input.getBlue() * .131));
-        setPixel(x, y, new Pixel(Math.min(outputRed, 255), Math.min(outputGreen, 255), Math.min(outputBlue, 255), 255));
+        int alpha = getPixel(x, y).getAlpha();
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/SepiaFilter.java]: same as GrayScaleFilter

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/SepiaFilter.java [hunk 7]
```diff
@@ -10,7 +10,8 @@ public void apply(int x, int y) {
...
+        setPixel(x, y, new Pixel(Math.min(outputRed, 255), Math.min(outputGreen, 255), Math.min(outputBlue, 255), alpha));
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/color/SepiaFilter.java]: we don't ready need to check Math.min anymore in filters since we moved it to setPixel

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/convolve/BlurFilter.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package pl.edu.uj.JImageStream.filters.convolve;
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/convolve/BlurFilter.java]: duplicate of BoxBlurFilter?

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/ConvolutionFilter.java [hunk 9]
```diff
@@ -28,8 +28,8 @@ public void apply(int x, int y) {
                 outputGreen += (int) ((neighbours.get(i).getGreen()) * kernel[i]);
                 outputBlue += (int) ((neighbours.get(i).getBlue()) * kernel[i]);
             }
-            setPixel(x, y, new Pixel(outputRed, outputGreen, outputBlue, 255));
-
+            int alpha = getPixel(x, y).getAlpha();
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/ConvolutionFilter.java]: is there any reason not to convolve alpha channel aswell?

Comment (Grebosz) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/ConvolutionFilter.java]: There is - when alpha channel is changed with the same algorithm as other channels, RobertsCross filter's produce non-visible content.

File: jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java [hunk 10]
```diff
@@ -7,9 +7,13 @@
 
 import java.awt.image.BufferedImage;
 import java.awt.image.WritableRaster;
+import java.lang.reflect.Array;
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java]: Why do we have these imports?

File: jis-api/src/main/java/pl/edu/uj/JImageStream/model/StreamableImage.java [hunk 11]
```diff
@@ -1,41 +1,35 @@
...
 
-    //todo same, maybe
+    //todo refactor, move to another class
     public void save(String format, String file) throws IOException {
         ImageIO.write(bufferedImage, format, new File(file));
     }
 
-    public int getHeight() {
-        return bufferedImage.getHeight();
+    public BufferedImage getBufferedImage() {
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/model/StreamableImage.java]: revert these changes, they are part of another issue

File: jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/RobertsCrossFilterTest.java [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
-import pl.edu.uj.JImageStream.filters.EdgeDetectionFilter;
-import pl.edu.uj.JImageStream.filters.GrayScaleFilter;
-import pl.edu.uj.JImageStream.filters.MedianFilter;
-import pl.edu.uj.JImageStream.filters.RobertsCrossXFilter;
-import pl.edu.uj.JImageStream.filters.RobertsCrossYFilter;
-
-/**
- * Created by kuba on 2016-11-13.
- */
+import pl.edu.uj.JImageStream.filters.edge.EdgeDetectionFilter;
```
Comment (jgprogramming) [inline jis-documentation/src/test/java/pl/edu/uj/JImageStream/tests/filters/RobertsCrossFilterTest.java]: fyi, ctrl + alt + o to organize your imports in inteliij

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MaxFilter.java [hunk 13]
```diff
@@ -21,21 +16,9 @@ public MaxFilter(int maskSize) {
...
-                try {
-                    pixelList.add(getPixel(x + i, y + j));
-                } catch (Exception e) {
-                    //corner case, no need to handle
-                }
-            }
-        }
+        List<Pixel> pixelList = getPixelList(x, y);
 
         Integer red = pixelList.stream().map(Pixel::getRed).sorted().max(Comparator.naturalOrder()).get();
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MaxFilter.java]: we could create abstract method in StatisticalFilter that would take List<Pixel> and return Pixel. This way all we have to override in max/mean/min/median filters would be this new method instead of whole apply logic. for reference, apply in statistical filter could look like this
apply(int x ,int y){
Pixel p = getPixelResult(getPixelList(x,y));
setPixel(x, y, p);
}

Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MaxFilter.java]: and getPixelResult would be implemented in each of statistical filters in a different way to return different pixel

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/ConvolutionFilter.java (hunk 9 above)
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/ConvolutionFilter.java]: are you sure? It works the same for me as it should. Could you provide an example?

File: jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MedianFilter.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    protected Pixel getPixelResult(List<Pixel> list) {
+        List<Integer> red = list.stream().map(Pixel::getRed).sorted().collect(Collectors.toList());
+        List<Integer> green = list.stream().map(Pixel::getGreen).sorted().collect(Collectors.toList());
+        List<Integer> blue = list.stream().map(Pixel::getBlue).sorted().collect(Collectors.toList());
+        List<Integer> alpha = list.stream().map(Pixel::getAlpha).sorted().collect(Collectors.toList());
+
+        return new Pixel(red.get(red.size() / 2),
```
Comment (jgprogramming) [inline jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/statistical/MedianFilter.java]: this isn't always median
